scraper/xscraper/factory.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package xscraper // import "go.opentelemetry.io/collector/scraper/xscraper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pipeline"
+	"go.opentelemetry.io/collector/scraper"
+)
+
+type Factory interface {
+	scraper.Factory
+
+	// CreateProfiles creates a Profiles scraper based on this config.
+	// If the scraper type does not support profiles,
+	// this function returns the error [pipeline.ErrSignalNotSupported].
+	CreateProfiles(ctx context.Context, set scraper.Settings, cfg component.Config) (Profiles, error)
+
+	// ProfilesStability gets the stability level of the Profiles scraper.
+	ProfilesStability() component.StabilityLevel
+}
+
+// FactoryOption apply changes to Options.
+type FactoryOption interface {
+	// applyOption applies the option.
+	applyOption(o *factory)
+}
+
+var _ FactoryOption = (*factoryOptionFunc)(nil)
+
+// factoryOptionFunc is a FactoryOption created through a function.
+type factoryOptionFunc func(*factory)
+
+func (f factoryOptionFunc) applyOption(o *factory) {
+	f(o)
+}
+
+type factory struct {
+	cfgType component.Type
+	component.CreateDefaultConfigFunc
+	scraper.Factory
+	createProfilesFunc     CreateProfilesFunc
+	profilesStabilityLevel component.StabilityLevel
+}
+
+func (f *factory) Type() component.Type {
+	return f.cfgType
+}
+
+func (f *factory) ProfilesStability() component.StabilityLevel {
+	return f.profilesStabilityLevel
+}
+
+func (f *factory) CreateProfiles(ctx context.Context, set scraper.Settings, cfg component.Config) (Profiles, error) {
+	if f.createProfilesFunc == nil {
+		return nil, pipeline.ErrSignalNotSupported
+	}
+	return f.createProfilesFunc(ctx, set, cfg)
+}
+
+// CreateProfilesFunc is the equivalent of Factory.CreateProfiles().
+type CreateProfilesFunc func(context.Context, scraper.Settings, component.Config) (Profiles, error)
+
+// WithProfiles overrides the default "error not supported" implementation for CreateProfiles and the default "undefined" stability level.
+func WithProfiles(createProfiles CreateProfilesFunc, sl component.StabilityLevel) FactoryOption {
```

Comment:
This is lacking tests.

Also, we need proxy methods to the stable signals here, or we can't mix stable and unstable together.
See https://github.com/open-telemetry/opentelemetry-collector/blob/receiver/xreceiver/v0.136.0/receiver/xreceiver/profiles.go#L97

Reply:
resolving with https://github.com/open-telemetry/opentelemetry-collector/pull/13915/commits/29c34fd8e2fa2e1995c3e050220ad75749e3d203